Kernel/Net/E1000ENetworkAdapter.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2018-2021, Andreas Kling <kling@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/MACAddress.h>
+#include <Kernel/Debug.h>
+#include <Kernel/Net/E1000ENetworkAdapter.h>
+#include <Kernel/PCI/IDs.h>
+
+namespace Kernel {
+
+#define REG_EEPROM 0x0014
+
+static bool is_valid_device_id(u16 device_id)
+{
+    switch (device_id) {
+    case 0x10D3: // 82574
+        return true;
+    default:
+        return false;
+    }
+}
+
+UNMAP_AFTER_INIT RefPtr<E1000ENetworkAdapter> E1000ENetworkAdapter::try_to_initialize(PCI::Address address)
+{
+    auto id = PCI::get_id(address);
+    if (id.vendor_id != (u16)PCIVendorID::Intel)
+        return {};
+    if (!is_valid_device_id(id.device_id))
+        return {};
+    u8 irq = PCI::get_interrupt_line(address);
+    auto adapter = adopt_ref(*new E1000ENetworkAdapter(address, irq));
```

Comment:
I'm not sure what the policy about doing OOM-unsafe stuff on init is, but this could also use the `adopt_ref_if_nonnull` treatment.